Firestore/core/src/firebase/firestore/remote/datastore.mm

```diff
@@ -38,6 +60,102 @@
 }
 
 }  // namespace
+
+const char *const kXGoogAPIClientHeader = "x-goog-api-client";
+const char *const kGoogleCloudResourcePrefix = "google-cloud-resource-prefix";
+
+Datastore::Datastore(util::AsyncQueue *firestore_queue,
```

Comment:
This parameter ordering is inconsistent with the other platforms. Admittedly there's not much consistency elsewhere but we should be working toward consistency, not away.

  * ObjC: databaseInfo, workerQueue, credentials
  * Android: databaseInfo, workerQueue, credentials
  * Typescript: workerQueue, connection, credentials, serializer

This should likely be: databaseInfo, workerQueue, credentials

(Also, this appears to be missing the credentials provider.)